### ISSUE TITLE
use HTTPS instead of HTTP for the GitHub Pages link?

### ISSUE THREAD

**Issue by E0735389** [nus-cs2103-AY2324S2/forum#382]
The textbook contains

> Go to `http://{your username}.github.io/{repo name}/` to view the user guide of your product.

https://nus-cs2103-ay2324s2.github.io/website/schedule/week6/project.html#a-userguide

I think it's better to use `https` instead (because GitHub will redirect to the HTTPS page anyway, but if the URL were HTTPS from the start we can avoid a man-in-the-middle attack)

**Comment by camille-readbean** [comment]
If you are worried about MITM attacks. You can turn on HTTPS only mode in your browser.  
  
Alternatively if you are setting up your own website next time, enable HSTS.  
  
If you are making web applications next time, make sure a link to your website does put sensitive parameters in your query parameters.  
  
TBH I think this is both a "client" and "server" issue? No one is stopping you from enforcing HTTPS only? And the website does not server anything particularly "sensitive" either. If you want a higher level of confidentiality and privacy perhaps Github pages should not be used to host your website. . .
  
I could see the case if the user was able to trigger a request through query parameters or whatever, e.g.  
http://mysite/makeUser/?userName=XXX&Password=XXXX  
But this link does not do it. Its also bad practice to make this kind of URLs anyway. (This form of URLs are common in verification emails where they need the url to pass content to the server and it opens up a can of worms like CSRF)  
Then perhaps yes HTTPS should be used. But in this case, HTTPS does not really serve a purpose?

TL;DR I think its more a nitpick... I give it a  
CVE 0.1

**Comment by damithc** [comment]
Thanks for posting this @E0735389 and for the analysis @camille-readbean 
I've changed it to https in any case.

**Comment by E0735389** [comment]
@camille-readbean Technically the issue with password in GET request parameter (and embedded in URL) is orthogonal with HTTP. If HTTP is used, MITM attackers can sniff passwords anyway even if they're sent through a POST request.
